src/Databases/DatabaseAtomic.cpp

```diff
@@ -0,0 +1,349 @@
+#include <Databases/DatabaseAtomic.h>
+#include <Databases/DatabaseOnDisk.h>
+#include <Poco/File.h>
+#include <IO/ReadHelpers.h>
+#include <IO/WriteHelpers.h>
+#include <Common/Stopwatch.h>
+#include <Parsers/formatAST.h>
+#include <Common/rename.h>
+#include <Storages/StorageMaterializedView.h>
+
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int UNKNOWN_TABLE;
+    extern const int TABLE_ALREADY_EXISTS;
+    extern const int CANNOT_ASSIGN_ALTER;
+    extern const int DATABASE_NOT_EMPTY;
+    extern const int NOT_IMPLEMENTED;
+}
+
+class AtomicDatabaseTablesSnapshotIterator final : public DatabaseTablesSnapshotIterator
+{
+public:
+    explicit AtomicDatabaseTablesSnapshotIterator(DatabaseTablesSnapshotIterator && base)
+        : DatabaseTablesSnapshotIterator(std::move(base)) {}
+    UUID uuid() const override { return table()->getStorageID().uuid; }
+};
+
+
+DatabaseAtomic::DatabaseAtomic(String name_, String metadata_path_, Context & context_)
+    : DatabaseOrdinary(name_, metadata_path_, "store/", "DatabaseAtomic (" + name_ + ")", context_)
```

Comment:
std::move for name_ and metadata_path_

Reply:
`name_` cannot be moved because it's also used in other argument